InstrumentCode.js

```diff
@@ -0,0 +1,104 @@
+goog.module('instrument.code');
+
+/** @record */
+class InstrumentationPerLine {
+  /**
+   * @param {!Array<number>} colNumberArray
+   * @param {!Array<number>} frequency
+   */
+  constructor(colNumberArray, frequency) {
+    /** @type {!Array<number>} */
+    this.colNumberArray = colNumberArray;
+    /** @type {!Array<number>} */
+    this.frequency = frequency;
+  }
+}
+
+class InstrumentCode {
+  constructor() {
+    /** @type {boolean} */
+    this.hasReportSent = false;
+    /** @type {!Map<string, !Map<number, !InstrumentationPerLine>>} */
+    this.reportData = new Map();
+  }
+
+  /**
+   * The instrumentCode function which is the instrumentation functon that will
+   * be called during program execution. It gathers all the data in the
+   * reportData variable defined above.
+   *
+   * @param {string} param
+   * @param {number} lineNumber
+   * @param {number} colNumber
+   */
+  instrumentCode(param, lineNumber, colNumber) {
+    if (this.hasReportSent) {
```

Comment:
We want to support periodic reports (as some JS will be executed later, e.g. when ad comes back from server).